pkg/curatedpackages/packageclient.go

```diff
@@ -46,3 +59,74 @@ func convertBundleVersionToPackageVersion(bundleVersions []packagesv1.SourceVers
 	}
 	return versions
 }
+
+func (pc *PackageClient) GeneratePackages() ([]packagesv1.Package, error) {
+	packageMap := pc.getPackageNameToPackage()
+	var packages []packagesv1.Package
+	for _, p := range pc.packages {
+		bundlePackage := packageMap[strings.ToLower(p)]
+		if bundlePackage.Name == "" {
+			fmt.Println(fmt.Errorf("unknown package %s", p).Error())
+			continue
+		}
+		name := CustomName + strings.ToLower(bundlePackage.Name)
+		packages = append(packages, convertBundlePackageToPackage(bundlePackage, name, pc.bundle.APIVersion))
+	}
+	return packages, nil
+}
+
+func (pc *PackageClient) WritePackagesToFile(packages []packagesv1.Package, d string) error {
+	directory := filepath.Join(d, packageLocation)
+	if !validations.FileExists(directory) {
```

Comment:
The command already aborts if the directory doesn't exist (https://github.com/aws/eks-anywhere/pull/1947/files#diff-7f848af9ae27534974e24287c26b1a19c58373183398dc3672d1ddaa2fa1d58fR55). So this check is superfluous.

Reply:
This appends curated-packages and checks the existence of this folder.